src/main/java/gov/nasa/pds/api/engineering/controllers/MyBundlesApiController.java

```diff
@@ -81,100 +73,54 @@ public MyBundlesApiController(ObjectMapper objectMapper, HttpServletRequest requ
-	    	
-	    	if (getBundleResponse.isExists()) {
-	    		MyBundlesApiController.log.info("get response " + getBundleResponse.toString());
-        		@SuppressWarnings("unchecked")
-				List<String> collections = (List<String>) getBundleResponse.getSourceAsMap().get("ref_lid_collection");
-        		String collectionLidVid;
-        		int i=0;
-        		for (String collectionLid : collections ) {
-        			if ((i>=start) && (i<start+limit)) {
-	        			collectionLidVid = this.productBO.getLatestLidVidFromLid(collectionLid);
-	        			
-	        			//this.productBO.getProduct(collectionLidVid);
-	        			GetRequest getCollectionRequest = new ElasticSearchRegistrySearchRequestBuilder().getGetProductRequest(collectionLidVid);
-	        			
-	        			//GetRequest getCollectionRequest = new GetRequest(this.esRegistryConnection.getRegistryIndex(), 
-	        			//		collectionLidVid);
-	        			
-	        			
-	        	        GetResponse getCollectionResponse = null;
-	        	        
-	        	        getCollectionResponse = restHighLevelClient.get(getCollectionRequest, 
-	                			RequestOptions.DEFAULT);
-	                	
-	    	        	if (getCollectionResponse.isExists()) {
-	    	        		
-	    	        		MyBundlesApiController.log.info("get response " + getCollectionResponse.toString());
-	    	        		Map<String, Object> sourceAsMap = getCollectionResponse.getSourceAsMap();
-	    	        		Map<String, XMLMashallableProperyValue> filteredMapJsonProperties = ProductBusinessObject.getFilteredProperties(sourceAsMap, fields, null);
-	
-	    	        		uniqueProperties.addAll(filteredMapJsonProperties.keySet());
-	
-	    	        		    	        	    
-	    	     	        
-	    	     	        if (!onlySummary) {
-	    	     	        	EntityProduct entityCollection = objectMapper.convertValue(sourceAsMap, EntityProduct.class);
-
-	    	     	        	Product collection = ElasticSearchUtil.ESentityProductToAPIProduct(entityCollection, this.getBaseURL());
-	    	     	        	collection.setProperties((Map<String, PropertyArrayValues>)(Map<String, ?>)filteredMapJsonProperties);
-	    	         	        products.addDataItem(collection);
-	    	     	        }
-	    	        		
-	    	        	
-	    	        	}
-	    	        	else {
-	    	        		MyBundlesApiController.log.warn("Couldn't get collection child " + collectionLidVid + " of bundle " + lidvid + " in elasticSearch");
-	    	        	}
-        			}
-        			i+=1; 
-        			
-        		}
-        	
-	    	}
-	    	
-	    	
-	    	summary.setProperties(new ArrayList<String>(uniqueProperties));
-	    	
-	    	return products;	
-			
-			
-		} catch (IOException e) {
-			  MyBundlesApiController.log.error("Couldn't get bundle " + lidvid + " from elasticSearch", e);
-              throw(e);
-		}
-    	
+
+    	HashSet<String> uniqueProperties = new HashSet<String>();
+    	List<String> clidvids = this.getRefLidCollection(lidvid, limit);
```

Comment:
Because you want products which are dereferenced from the collection so that start is not meaningful. For instance bundle foo has 10 collections with each collection having 500 products. If the start is 5 and you start at the 5th collection, then you missed 500*5 - 5 products.

Limit is used to help optimize the iteration. Could have it default to anything but though that using the limit of the call was a way to allow the user to optimize the data retrieval. Not really sure it will work but that is the idea.